### ISSUE TITLE
Error when selecting row in grid with hidden column

### ISSUE THREAD

**Issue by anderscore-hjhessmann** [vaadin/flow-components#4476]
### Description

Create a Grid with grid.setSelectionMode(Grid.SelectionMode.MULTI) and hide a column with setVisible(false) then the following error occurs:

There seems to be an error in Vaadin Grid:
row is null
Please submit an issue to https://github.com/vaadin/flow-components/issues/new/choose [localhost:8080:7:224](http://localhost:8080/)
    error http://localhost:8080/:7
    tryCatchWrapper http://localhost:8080/VAADIN/generated/jar-resources/FlowBootstrap.js:67
    _onSelectClick http://localhost:8080/VAADIN/generated/jar-resources/vaadin-grid-flow-selection-column.js:131

### Expected outcome

I'd expect the row to be selected as with Vaadin 23.3.0. The bug was introduced in Vaadin 23.3.1

### Minimal reproducible example

```
public class GridView extends VerticalLayout {

    public GridView() {
        setSpacing(false);

        Grid<SampleData> grid = new Grid<>();
        grid.addColumn(SampleData::col1).setHeader("Col. A").setResizable(true);
        Grid.Column<SampleData> columnB = grid.addColumn(SampleData::col2).setHeader("Col. B").setResizable(true);
        grid.setSelectionMode(Grid.SelectionMode.MULTI);

        Checkbox cb = new Checkbox("Show Column B");
        cb.setValue(true);
        cb.addValueChangeListener(ev -> columnB.setVisible(ev.getValue()));

        grid.setItems(IntStream.rangeClosed(1, 9).mapToObj(SampleData::new).toList());
        add(grid, cb);
    }

    record SampleData(
            String col1,
            String col2
    ) {
        SampleData(int no) {
            this("A" + no, "B" + no);
        }
    }
}

```

### Steps to reproduce

1. Select some rows
2. De-Select 'Show Column B'
3. Select or de-select more rows

Step 3 will fail.

### Environment

Vaadin version(s): 23.3.1
OS: Windows 10

### Browsers

Chrome, Firefox

**Comment by web-padawan** [comment]
Thanks for the issue. This is most likely a duplicate of https://github.com/vaadin/flow-components/issues/4429 fixed by https://github.com/vaadin/web-components/pull/5153. 
The fix was released in Vaadin 23.3.2. Could you please try if upgrading fixes the issue?

**Comment by anderscore-hjhessmann** [comment]
Yes, 23.3.2 has fixed the issue. Thank you.
